src/unix/poll.c

```diff
@@ -33,8 +33,19 @@ static void uv__poll_io(uv_loop_t* loop, uv__io_t* w, unsigned int events) {
 
   handle = container_of(w, uv_poll_t, io_watcher);
 
-  if (events & POLLERR) {
-    uv__io_stop(loop, w, POLLIN | POLLOUT | UV__POLLRDHUP);
+  /*
+   * As documented in the kernel source fs/kernfs/file.c #780
+   * poll will return POLLERR|POLLPRI in case of sysfs
+   * polling. This does not happen in case of out-of-band
+   * TCP messages.
+   *
+   * The above is the case on (at least) FreeBSD and Linux.
+   *
+   * So to properly determine a POLLPRI or a POLLERR we need
+   * to check for both.
+   */
+  if ((events & POLLERR) && !(events & UV__POLLPRI)) {
```

Comment:
I realize I am reviewing a very old PR, but I noticed that in the comment here, you check for `POLLERR && !POLLPRI` as a distinctive condition for linux, but in this same PR, you changed the epoll backend to unconditionally add POLLPRI if the event was POLLERR alone:

```diff
      if (pe->events == POLLERR || pe->events == POLLHUP)
-         pe->events |= w->pevents & (POLLIN | POLLOUT);
+         pe->events |= w->pevents & (POLLIN | POLLOUT | UV__POLLPRI);
```

Am I confused or is this broken?

Reply:
Except for the theoretical issue, did you encounter an actual practical issue with this possible inconsistancy?

Reply:
Except for the theoretical issue, did you encounter an actual practical issue with this possible inconsistancy?

Reply:
No, I was just noticing the added comments appeared to be wrong here

Reply:
Can you suggest changes to make it consistant. The comment about the kernel is what happens, and i believe the bitwise change is also what is required to make it work properly.

Reply:
The linux-core change seemed unnecessary

Reply:
That can easily be verified by removing it, and running the unittests again.